### DISCUSSION TITLE
Any tips on how I can reconcile parts "produced" by other parts?

### DISCUSSION THREAD

**Discussion by Guusggg** [inventree/InvenTree#2443] [General]
Hi,

Today I came across a "philosophical" question about how to reconcile when a part is produced by other parts.
The reality is that we order one tube of 6 meters. I want to specify to InvenTree that, in order to produce a tube of 1 meter, it will consume one 6m tube to create 6 1m tubes.

I tried using build orders, but they can only do the inverse. 6 6m tubes would make 6 1m tubes.

Is this possible to reconcile, or do you have any tips on how I can handle this effectively?

(@SchrodingersGat already gave me an answer but I didn't quite get it. The units of my InvenTree instance are `string`s so I don't see how it helps with getting InvenTree to understand I can cut pieces of 1m out of 6m tubes.)

**InvenTree Version | 0.5.3**

Edit: Would Part Parameters help in this situation?

Kind regards,

**Comment by matmair** [comment]
@Guusggg I would recommend saving the tube as stock in meters and using the part in the BOM accordingly also in meters. So if you need a 1-meter section the quantity would be 1, if you need a 75 centimetres section the quantity would be 0.75 and similar for freedom units - but I suggest SI.
If you then create a build order you can allocate the fractional amount. This only works with not serialized items as those only support a quantity of 1.

**Reply by eeintech** [reply]
Also your 1m tube can be an assembly which gets the 6m tube in its BoM with quantity = 1/6 = 0.1667
So for instance if you make 3x 1m tubes the build order would consume half a 6m tube

**Comment by Guusggg** [comment]
Thanks for the comments guys! Now I'm wondering if I can specify a "standard" stock level for parts. Like if I buy a 11m tube from a purchase order, I want it to add 11 units to the 1m part. I haven't found this yet, am I overlooking something?

**Reply by Guusggg** [reply]
Well, I thought about it for some time and I did manage to do it by:
1. Create a new part with units set to "meters" (Tube)
2. Create a new part with units set to nothing (6m Tube)
3. Add BOM to Tube with 1/6 = 0.16667 as only BOM Line Item

**Reply by matmair** [reply]
@Guusggg glad you found a solution! Just my 2 $: Why not use standard metric (or imperial) lengths instead of 1/6 of a 6 meter part? This has the potential to confuse users.
Also are you happy with this solution or do you have suggestions for a better workflow?

**Reply by Guusggg** [reply]
@matmair Thank you for thinking with me! The reason why I use `1/6` is because the part is being delivered as 6m tube, which we cut into pieces. If I want to put them back into "units", then our purchasing offices needs to input the length of the tube in the **Purchase Order**, which I find too error-prone for humans. If our purchasing office needs to buy 4 tubes of 6m, they need to be able to input 4 parts of 6m. Not do any calculation on their own for the correct length (4 * 6 (meters) * 1000 (mm)). This relates to the fact that **currently, I haven't found a way to map 1 part of 6m into 6000 parts of 1mm WITHOUT using Build Orders.**

Here's what I want to do:
We have 1 part, 6m long that we cut pieces of. I would love to turn this 6m part into its unit form, in millimeters, so we now how many millimeters we have left. If we want to cut a 50cm piece, we would take 500 this from the "unit part". Same for 1 meter pieces, but 100. With this "unit part", I can let these be pieces tracked individually and we know exactly how many **pieces** of how much **length** we have left over so it's possible to optimize our inventory, and not cut another 6m tube but maybe reuse a piece of 1m that was erroneously cut. It would also allow us to cut tubing of custom length for one specific order. (Also, if all parts are linked together through this "millimeter unit part", all prices will be automagically be taken from how much 1mm cost for us)

Sorry for the long explanation and I think I'm trying to make this too generic. I've mostly thrown the generic "millimeter unit part" out of the window for now, as it would require making at least 2 build orders per cut (1 Build Order to make the 6m part into 600 "unit parts", and 1 Build Order to make the 100 "unit parts" into 1m pieces)

I hope I'm just massively overseeing something lol my head is exploding from this stupid complexity that I engineered myself XD I feel like a painter, painting himself into a corner

(also a stupid question but I haven't found a way to message you privately, but would you mind checking out the pull request I put in at the F360 InvenTreeLink repo? I'm pretty proud of the solution I came up with to synchronize all parts in F360 with InvenTree)

**Reply by Guusggg** [reply]
@matmair There was a public InvenTree demo somewhere right? Does it reset often? Maybe I can make a proof of concept there, and you could check out what I mean more precisely ^^

**Reply by SchrodingersGat** [reply]
https://demo.inventree.org

**Reply by Guusggg** [reply]
@SchrodingersGat @matmair Hi, I made a really convulated way of doing what I want to do. Would you mind checking it out if it can be optimized? https://demo.inventree.org/part/category/21/#

**Reply by matmair** [reply]
The instance resets daily so we probably need another method to discuss this

**Comment by Guusggg** [comment]
Okay this is how I solved it for now:

1. Create a new part with units set as "meters" (Tube)
2. Create a new part with units set to nothing (1m Tube)
3. Add BOM to 1m Tube as needing 1 "meters" Tube
4. Create Build Order of 1m Tube correctly subtracts 1 "meters" from Tube

I will follow up with how I'm going to do one-way conversion.
- I want 1m Tube to be 'built' into 2x 50cm Tube, but not the other way around.

**Reply by eeintech** [reply]
@Guusggg Want to mark it as answer?

**Reply by Guusggg** [reply]
Oops, my bad, Will do!

**Reply by Guusggg** [reply]
Hmm I can't seem to find where...
